types/sortable-options.d.ts

```diff
@@ -0,0 +1,204 @@
+	 */
+	animation?: number;
+	/**
+	 * Class name for the chosen item
+	 */
+	chosenClass?: string;
+	dataIdAttr?: string;
+	/**
+	 * time in milliseconds to define when the sorting should start
+	 */
+	delay?: number;
+	/*
+	 * only delay if user is using touch
+	 */
+	delayOnTouchOnly?: Boolean;
+	/**
+	 * Direction of Sortable (will be detected automatically if not given)
+	 */
+	direction?: 'horizontal' | 'vertical' | ((event: Event, target: HTMLElement, dragEl: HTMLElement) => 'horizontal'|'vertical');
+	/**
+	 * Disables the sortable if set to true.
+	 */
+	disabled?: boolean;
+	/**
+	 * Class name for the dragging item
+	 */
+	dragClass?: string;
+	/**
+	 * Specifies which items inside the element should be draggable
+	 */
+	draggable?: string;
+	dragoverBubble?: boolean;
+	dropBubble?: boolean;
+	/**
+	 * Easing for animation. Defaults to null. See https://easings.net/ for examples.
+	 */
+	easing?: string;
+	/**
+	 * distance in px mouse must be from empty sortable to insert drag element into it
+	 */
+	emptyInsertThreshold?: number;
+	/**
+	 * Class name for the cloned DOM Element when using forceFallback
+	 */
+	fallbackClass?: string;
+	/**
+	 * Appends the cloned DOM Element into the Document's Body
+	 */
+	fallbackOnBody?: boolean;
+	/**
+	 * Specify in pixels how far the mouse should move before it's considered as a drag.
+	 */
+	fallbackTolerance?: number;
+	fallbackOffset?: { x: number, y: number };
+	/**
+	 * Selectors that do not lead to dragging (String or Function)
+	 */
+	filter?: string | ((this: Sortable, event: Event | TouchEvent, target: HTMLElement, sortable: Sortable) => boolean);
+	/**
+	 * ignore the HTML5 DnD behaviour and force the fallback to kick in
+	 */
+	forceFallback?: boolean;
+	/**
+	 * Class name for the drop placeholder
+	 */
+	ghostClass?: string;
+	/**
+	 * To drag elements from one list into another, both lists must have the same group value.
+	 * You can also define whether lists can give away, give and keep a copy (clone), and receive elements.
+	 */
+	group?: string | SortableGroupOptions;
+	/**
+	 * Drag handle selector within list items
+	 */
+	handle?: string;
+	ignore?: string;
+	/**
+	 * Threshold of the inverted swap zone (will be set to swapThreshold value by default)
+	 */
+	invertedSwapThreshold?: number,
+	/**
+	 * Will always use inverted swap zone if set to true
+	 */
+	invertSwap?: boolean,
+	/**
+	 * Call `event.preventDefault()` when triggered `filter`
+	 */
+	preventOnFilter?: boolean;
+	/**
+	 * Remove the clone element when it is not showing, rather than just hiding it
+	 */
+	removeCloneOnHide?: boolean;
+	scroll?: boolean;
+	/**
+	 * if you have custom scrollbar scrollFn may be used for autoscrolling
+	 */
+	scrollFn?: ((this: Sortable, offsetX: number, offsetY: number, event: MouseEvent) => void);
+	/**
+	 * px, how near the mouse must be to an edge to start scrolling.
+	 */
+	scrollSensitivity?: number;
+	/**
+	 * px
+	 */
+	scrollSpeed?: number;
```

Comment:
These are all part of the `AutoScroll` plugin, is that correct?

If so, they should be in a separate interface and have `SortbaleOptions` extend them.

```ts
export interface SortableOptions extends AutoScrollOptions {
 // ... the options
}

```